### PR TITLE
Switch from AddUntimed to AddTimed for ResendEnabled

### PR DESCRIPTION
**Pull request by ryanhall07**
This allows for a graceful transition from untimed -> timed metrics.

Previously when the coordinator would switch from untimed -> timed, it
would result in a new aggregation for that metric on the leader. That
incomplete aggregation would get written by the leader, resulting in
incorrect aggregated data. An example is a prom counter that is not
increasing. The new aggregation starts at 0 and the prom counter might
be at value 1000. Even though the counter hasn't increased, the
aggregator will output an increase of 1000 for the first aggregation,
which is incorrect.

Now the transition happens in the aggregator. When the follower is
deployed with the feature flag enabled, it will migrate untimed ->
timed. Once the follower is ready to take over as the leader, it will
have complete timed aggregations to start flushing.

This is a very similar problem for deploys. The first metrics written to
the follower after a deploy are incomplete, which is why we need to wait
for the bufferDurationForPastTimed metrics for the follower to takeover.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Review by wesleyk — APPROVED** [review]

**Review by robskillington — APPROVED** [review]
LGTM

File: src/aggregator/aggregator/aggregator.go [hunk 1]
```diff
@@ -219,20 +220,39 @@ func (agg *aggregator) placementTick() {
 }
 
 func (agg *aggregator) AddUntimed(
-	metric unaggregated.MetricUnion,
+	union unaggregated.MetricUnion,
 	metadatas metadata.StagedMetadatas,
 ) error {
+	if agg.opts.TimedForResendEnabled() {
+		for _, m := range metadatas {
+			for _, p := range m.Pipelines {
```
Comment (vdarulis) [inline src/aggregator/aggregator/aggregator.go:228]:
```suggestion
	if agg.opts.TimedForResendEnabled() {
		for i := range metadatas {
			for j := range m[i].Pipelines {
			...
```

Comment (vdarulis) [reply src/aggregator/aggregator/aggregator.go:228]: these structs can be quite big, I suggest iterating over indexes and not copying over values in each loop

File: src/aggregator/aggregator/aggregator.go [hunk 2]
```diff
@@ -219,20 +220,39 @@ func (agg *aggregator) placementTick() {
...
+				if p.ResendEnabled {
```
Comment (vdarulis) [inline src/aggregator/aggregator/aggregator.go:229]:
```suggestion
				if !p.ResendEnabled {
				    continue
				}
```

File: src/aggregator/aggregator/aggregator.go (hunk 1 above)
Comment (ryanhall07) [reply src/aggregator/aggregator/aggregator.go:228]: good call!

**Pull request merged**